tests/phpunit/tests/rest-api/wpRestTemplatesController.php

```diff
@@ -208,6 +208,119 @@ public function get_template_endpoint_urls() {
 		);
 	}
 
+	/**
+	 * @dataProvider data_get_item_with_valid_theme_dirname
+	 * @covers WP_REST_Templates_Controller::get_item
+	 * @ticket 54596
+	 *
+	 * @param string $theme_dir Theme directory to test.
+	 * @param string $template  Template to test.
+	 * @param array $args       Arguments to create the 'wp_template" post.
+	 */
+	public function test_get_item_with_valid_theme_dirname( $theme_dir, $template, array $args ) {
+		wp_set_current_user( self::$admin_id );
+		switch_theme( $theme_dir );
+
+		// Set up template post.
+		$args['post_type'] = 'wp_template';
+		$args['tax_input'] = array(
+			'wp_theme' => array(
+				get_stylesheet(),
+			),
+		);
+		$post              = self::factory()->post->create_and_get( $args );
+		wp_set_post_terms( $post->ID, get_stylesheet(), 'wp_theme' );
+
+		$request  = new WP_REST_Request( 'GET', "/wp/v2/templates/{$theme_dir}//{$template}" );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+		unset( $data['content'] );
+		unset( $data['_links'] );
+
+		$this->assertSame(
```

Comment:
```suggestion
		$this->assertSameSetsWithIndex(
```
Should this have a stricter assertion to help us spot side effects from future changes?

Reply:
Fixed in https://github.com/WordPress/wordpress-develop/pull/2064/commits/18299f81106d088be671d2ea75a7843141436a80